### ISSUE TITLE
console.assert Error running on Atom editor

### ISSUE THREAD

**Issue by gabrielboliveira** [vuejs/eslint-plugin-vue#35]
Hello guys! I've been successfully using this plugin with my Vue project directly on CLI, without any issues. 

Today I was able to make the linter work on Atom editor with .vue files, but there's an issue with `console.assert` calls on [lib/utils/index.js](https://github.com/vuejs/eslint-plugin-vue/blob/22aeca5c3ad2a5f5775b7a6134b56236de00b217/lib/utils/index.js). Looks like it doesn't have  `console.assert` method for some reason.

![console-assert-issue](https://user-images.githubusercontent.com/11093090/27265570-6258a7f6-546e-11e7-8402-a61a7d2cebb9.png)

If I remove those calls from the `node_modules` file locally, it works like a charm. So, questions: 

1. Are those really necessary?
2. Is this an `linter` (atom package) or perhaps Atom issue?

macOS 10.12.5
Atom 1.18.0
Atom Plugins: 
linter 2.2.0
linter-eslint 8.2.1

**Comment by mysticatea** [comment]
Thank you for this issue.

Hm, probably we can replace the assertions by `assert` module of Node.js core in order to solve this issue.

----

1. It's better if the assertions exist in order to develop this plugin.
2. At least, `console.assert` exists on all popular browsers (including old IE) and Node.js. Also, VSCode which is based on Electron seems no problem. However, `console` object is not standard. It might be better that we don't assume  it to work...

**Comment by gabrielboliveira** [comment]
Great! Liked your approach. 

Perhaps it's an package issue as well. I agree with you that it should have the `console.assert` method. I'll open an issue there also.

**Comment by Bartheleway** [comment]
As a quick fix you can open `node_modules/eslint-plugin-vue/lib/utils/index.js` and remove all the `console.assert`. I tried and this fixed the issue.

**Comment by gabrielboliveira** [comment]
@Bartheleway Yes, that works. I'm using this to be able to use the linter on Atom

**Comment by gabrielboliveira** [comment]
So, I've done some more testing, and the only `console` method I was able to use was `console.log`. 

Not sure why other methods are not available, but it's not an `eslint-plugin-vue` issue for sure, perhaps an improvement not trusting on `console.assert` being available.

**Comment by michalsnik** [comment]
Fix released as `v3.1.3` 🚀 It's funny though - in order to make eslint work on `.vue` files in atom you need to set `Lint HTML files` in `linter-eslint` atom's package and install `eslint-plugin-html`, but you don't need to include it in `.esllintrc.*` anywhere.

**Comment by gabrielboliveira** [comment]
@michalsnik Thanks!!

I was able to make it work only with `Lint HTML files` enabled, installed eslint on project and not globally. 

Then my `.eslintrc.json` looks like the default config:

```json
{
  "extends": [
    "eslint:recommended",
    "plugin:vue/recommended"
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  }
}
```
